### PR TITLE
feat: release ENS names

### LINKED ISSUE #782: Add button to release status ENS usernames after one year of being registered

### PR DESCRIPTION
**Pull request by richard-ramos**
It probably makes sense to test this after https://github.com/status-im/status-desktop/pull/3186 is merged.

Fixes #782

### CODE REVIEW

**Review by pinebit — APPROVED** [review]

File: src/status/ens.nim [hunk 1]
```diff
@@ -172,6 +173,54 @@ proc getPrice*(): Stuint[256] =
...
+    ensUsernamesContract = contracts.getContract("ens-usernames")
+    release = Release(label: label)
+
+  var tx = transactions.buildTokenTransaction(parseAddress(address), ensUsernamesContract.address, "", "")
+  try:
+    let response = ensUsernamesContract.methods["release"].estimateGas(tx, release, success)
+    if success:
+      result = fromHex[int](response)
+  except RpcException as e:
+    raise
```
Comment (alaibe) [inline src/status/ens.nim:188]: Is it usefull to try/except without doing anything in the except block but raise, I'd assume it just re-raise the same exception catch.

Also the exception should probably be catch in the client and handle?

File: src/status/ens.nim [hunk 2]
```diff
@@ -172,6 +173,54 @@ proc getPrice*(): Stuint[256] =
...
+    ensUsernamesContract = contracts.getContract("ens-usernames")
+    release = Release(label: label)
+
+  var tx = transactions.buildTokenTransaction(parseAddress(address), ensUsernamesContract.address, "", "")
+  try:
+    result = ensUsernamesContract.methods["release"].send(tx, release, password, success)
+    if success:
+      trackPendingTransaction(result, address, $ensUsernamesContract.address, PendingTransactionType.ReleaseENS, username)
+  except RpcException as e:
+    raise
```
Comment (alaibe) [inline src/status/ens.nim:202]: same question :)

File: src/status/wallet.nim [hunk 3]
```diff
@@ -91,7 +91,7 @@ proc confirmTransactionStatus(self: WalletModel, pendingTransactions: JsonNode,
     if transactionReceipt.kind != JNull:
       status_wallet.deletePendingTransaction(trx["hash"].getStr)
       let ev = TransactionMinedArgs(
-                data: trx["data"].getStr,
+                data: trx["additionalData"].getStr,
```
Comment (alaibe) [inline src/status/wallet.nim:94]: I also have this fix in another PR ;)

**Review by alaibe — APPROVED** [review]
LGTM, just a couple nit about error handling

**Comment by anastasiyaig** [comment]
@richard-ramos hey there, can we please rebase this so i can retest? thank you!

**Comment by richard-ramos** [comment]
>  hey there, can we please rebase this so i can retest? thank you!

done!

**Comment by anastasiyaig** [comment]
Screenshot 2021-08-30 at 22 35 34


i can't really test "release" functionality even on ropsten because the ReleaseDate parameter is hardcoded to 1 year to hold the name, and it could not be changed as i got from Ricardo and Richard

**Pull request merged**